packager/VisibilityChecker.cc

```diff
@@ -445,15 +445,45 @@ class VisibilityCheckerPass final {
                 auto &pkg = ctx.state.packageDB().getPackageInfo(otherPackage);
                 bool isTestImport = otherFile.data(ctx).isPackagedTest() || ctx.file.data(ctx).isPackagedTest();
                 auto strictDepsLevel = this->package.strictDependenciesLevel();
-                bool layeringViolation =
-                    !isTestImport && db.enforceLayering() && strictDepsLevel.has_value() &&
-                    strictDepsLevel.value().first != core::packages::StrictDependenciesLevel::False &&
-                    this->package.causesLayeringViolation(db, pkg);
-                if (layeringViolation) {
+                auto importStrictDepsLevel = pkg.strictDependenciesLevel();
+                bool layeringViolation = false;
+                bool strictDependenciesTooLow = false;
+                if (!isTestImport && db.enforceLayering()) {
+                    layeringViolation =
+                        strictDepsLevel.has_value() &&
+                        strictDepsLevel.value().first != core::packages::StrictDependenciesLevel::False &&
+                        this->package.causesLayeringViolation(db, pkg);
+                    strictDependenciesTooLow =
+                        importStrictDepsLevel.has_value() &&
+                        importStrictDepsLevel.value().first < this->package.minimumStrictDependenciesLevel();
+                }
+                // TODO(neil): Provide actionable advice and/or link to a doc that would help the user resolve these
+                // layering/strict_dependencies issues.
```

Comment:
I will do this after implementing the third case (don't suggest an import would cause a cycle for layered_dag/dag pacakges); I suspect there’ll be commonalities in the advice so I figure it’s easiest to write at the end.